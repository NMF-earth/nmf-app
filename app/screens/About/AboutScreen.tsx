import React from "react";
import { ScrollView, View } from "react-native";

import { Text, StickersImage } from "components";

import { t } from "utils";

import styles from "./AboutScreen.styles";
import navigationOptions from "./AboutScreen.navigationOptions";

const AboutScreen = () => {
  return (
    <ScrollView style={styles.container}>
      <StickersImage sticker="earth" />
      <Text.Primary style={styles.header}>
        {t("ABOUT_SCREEN_INTRO")}
      </Text.Primary>
      <Text.H2 style={styles.header}>{t("ABOUT_SCREEN_CARE_HEADER")}</Text.H2>
      <Text.H3 style={styles.subHeader}>{t("ABOUT_SCREEN_PRIVACY")}</Text.H3>
      <Text.Primary>{t("ABOUT_SCREEN_PRIVACY_BODY")}</Text.Primary>
      <Text.H3 style={styles.subHeader}>{t("ABOUT_SCREEN_ETHIC")}</Text.H3>
      <Text.Primary>{t("ABOUT_SCREEN_ETHIC_BODY")}</Text.Primary>
      <Text.H3 style={styles.subHeader}>
        {t("ABOUT_SCREEN_OPEN_SOURCE")}
      </Text.H3>
      <Text.Primary>{t("ABOUT_SCREEN_OPEN_SOURCE_BODY")}</Text.Primary>
      <Text.H2 style={styles.header}>
        {t("ABOUT_SCREEN_LIBRARIES_AND_CONTRIBUTORS")}
      </Text.H2>
      <View style={styles.githubView}>
        <Text.Primary>{t("ABOUT_SCREEN_CAN_BE_FOUND")}</Text.Primary>
        <Text.Link url="https://github.com/NMF-earth/nmf-app">
          {t("ABOUT_SCREEN_GITHUB")}
        </Text.Link>
      </View>
      <View style={styles.separator} />
    </ScrollView>
  );
};

AboutScreen.navigationOptions = navigationOptions;

export default AboutScreen;
