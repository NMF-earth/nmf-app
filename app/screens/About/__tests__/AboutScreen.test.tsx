import React from "react";
import renderer from "react-test-renderer";
import AboutScreen from "../AboutScreen";

it("AboutScreen renders correctly", () => {
  const tree = renderer.create(<AboutScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
