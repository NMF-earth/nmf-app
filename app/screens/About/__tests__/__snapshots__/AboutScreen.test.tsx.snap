// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <Text.Primary
      style={
        Object {
          "paddingVertical": 26,
        }
      }
    >
      We are a small team concerned by the human impact on the planet. We are not a startup driven by growth or profit to pay dividends to investors. We are transparent and you can have a look to our figures on Indie Hackers. You can join our team if you wanna help the project. Or simply talk around you about our project, give us 5 stars on the store et send us your feedback on the app will help us a lot. Thanks you ✌️
    </Text.Primary>
    <Text.H2
      style={
        Object {
          "paddingVertical": 26,
        }
      }
    >
      We care about you
    </Text.H2>
    <Text.H3
      style={
        Object {
          "paddingVertical": 12,
        }
      }
    >
      Privacy by design
    </Text.H3>
    <Text.Primary>
      All data provided in the app will never be sold or shared to any companies and we don't track what you do in the app 
    </Text.Primary>
    <Text.H3
      style={
        Object {
          "paddingVertical": 12,
        }
      }
    >
      Ethic by design
    </Text.H3>
    <Text.Primary>
      We don't exploit psychological vulnerabilities like bottomless feeds or auto-play to keep users from leaving 
    </Text.Primary>
    <Text.H3
      style={
        Object {
          "paddingVertical": 12,
        }
      }
    >
      100% Open source
    </Text.H3>
    <Text.Primary>
      Everyone will be to add content to the app, verify emissions models and fix any incorrect information 
    </Text.Primary>
    <Text.H2
      style={
        Object {
          "paddingVertical": 26,
        }
      }
    >
      Librairies & contributors
    </Text.H2>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text.Primary>
        Can be found on 
      </Text.Primary>
      <Text.Link
        url="https://github.com/NotMyFaultEarth/nmf-app"
      >
        Github
      </Text.Link>
    </View>
    <View
      style={
        Object {
          "height": 30,
        }
      }
    />
  </View>
</RCTScrollView>
`;
